### PR TITLE
hpp-fcl -> coal: 2.4.5 -> 3.0.0

### PR DESCRIPTION
**Pull request by nim65s**
Hi,

For version 3.0.0, hpp-fcl was renamed coal.

ref. https://github.com/coal-library/coal/releases/tag/v3.0.0

While here, update jrl-cmakemodules, as a new ros2 thing is required.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build pinocchio

**Comment by nim65s** [comment]
rebased after https://github.com/NixOS/nixpkgs/pull/358668

**Review by wegank — APPROVED** [review]

**Comment by nim65s** [comment]
Not sure for how long we should wait for pinocchio on aarch64-darwin :/

File: pkgs/by-name/co/coal/package.nix [hunk 1]
```diff
@@ -15,15 +16,14 @@
...
 
   src = fetchFromGitHub {
-    owner = "humanoid-path-planner";
-    repo = "hpp-fcl";
-    rev = "v${finalAttrs.version}";
-    fetchSubmodules = true;
-    hash = "sha256-0OORdtT7vMpvK3BPJvtvuLcz0+bfu1+nVvzs3y+LyQw=";
+    owner = "coal-library";
+    repo = "coal";
+    rev = "refs/tags/v${finalAttrs.version}";
```
Comment (GaetanLepage) [inline pkgs/by-name/co/coal/package.nix:25]:
```suggestion
    tag = "v${finalAttrs.version}";
```
You can use the new `tag` argument now. Not a blocker though.

File: pkgs/by-name/co/coal/package.nix [hunk 2]
```diff
@@ -74,8 +79,8 @@ stdenv.mkDerivation (finalAttrs: {
   '';
 
   meta = {
-    description = "Extension of the Flexible Collision Library";
-    homepage = "https://github.com/humanoid-path-planner/hpp-fcl";
+    description = "Collision Detection Library, previously hpp-fcl";
+    homepage = "https://github.com/coal-library/coal";
     license = lib.licenses.bsd3;
```
Comment (GaetanLepage) [inline pkgs/by-name/co/coal/package.nix:84]:
```suggestion
    homepage = "https://github.com/coal-library/coal";
    changelog = "https://github.com/coal-library/coal/releases/tag/${src.tag}";
    license = lib.licenses.bsd3;
```

Comment (nim65s) [reply pkgs/by-name/co/coal/package.nix:84]: I guess you mean `finalAttrs.src.tag` ?

**Review by LeSuisse — APPROVED** [review]
Changes are looking good.

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357705`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 93 packages built:</summary>
  <ul>
    <li>aligator</li>
    <li>aligator.doc</li>
    <li>casadi</li>
    <li>coal</li>
    <li>coal.dev</li>
    <li>coal.doc</li>
    <li>crocoddyl</li>
    <li>crocoddyl.doc</li>
    <li>eiquadprog</li>
    <li>eiquadprog.doc</li>
    <li>gepetto-viewer (python312Packages.gepetto-viewer)</li>
    <li>gepetto-viewer-corba (python312Packages.gepetto-viewer-corba)</li>
    <li>gepetto-viewer-corba.doc (python312Packages.gepetto-viewer-corba.doc)</li>
    <li>gepetto-viewer.bin (python312Packages.gepetto-viewer.bin)</li>
    <li>gepetto-viewer.dev (python312Packages.gepetto-viewer.dev)</li>
    <li>gepetto-viewer.doc (python312Packages.gepetto-viewer.doc)</li>
    <li>jrl-cmakemodules</li>
    <li>mim-solvers</li>
    <li>ndcurves</li>
    <li>ndcurves.doc</li>
    <li>pinocchio</li>
    <li>pinocchio.doc</li>
    <li>proxsuite</li>
    <li>proxsuite-nlp</li>
    <li>proxsuite-nlp.doc</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.aerosandbox</li>
    <li>python311Packages.aerosandbox.dist</li>
    <li>python311Packages.aligator</li>
    <li>python311Packages.aligator.doc</li>
    <li>python311Packages.casadi</li>
    <li>python311Packages.coal</li>
    <li>python311Packages.coal.dev</li>
    <li>python311Packages.coal.doc</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.crocoddyl.doc</li>
    <li>python311Packages.eigenpy</li>
    <li>python311Packages.eigenpy.dev</li>
    <li>python311Packages.eigenpy.doc</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.example-robot-data.doc</li>
    <li>python311Packages.gepetto-gui</li>
    <li>python311Packages.gepetto-viewer</li>
    <li>python311Packages.gepetto-viewer-corba</li>
    <li>python311Packages.gepetto-viewer-corba.doc</li>
    <li>python311Packages.gepetto-viewer.bin</li>
    <li>python311Packages.gepetto-viewer.dev</li>
    <li>python311Packages.gepetto-viewer.doc</li>
    <li>python311Packages.mim-solvers</li>
    <li>python311Packages.ndcurves</li>
    <li>python311Packages.ndcurves.doc</li>
    <li>python311Packages.neuralfoil</li>
    <li>python311Packages.neuralfoil.dist</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.pinocchio.doc</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite-nlp</li>
    <li>python311Packages.proxsuite-nlp.doc</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.tsid</li>
    <li>python311Packages.tsid.doc</li>
    <li>python312Packages.aerosandbox</li>
    <li>python312Packages.aerosandbox.dist</li>
    <li>python312Packages.aligator</li>
    <li>python312Packages.aligator.doc</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.coal</li>
    <li>python312Packages.coal.dev</li>
    <li>python312Packages.coal.doc</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.crocoddyl.doc</li>
    <li>python312Packages.eigenpy</li>
    <li>python312Packages.eigenpy.dev</li>
    <li>python312Packages.eigenpy.doc</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.example-robot-data.doc</li>
    <li>python312Packages.gepetto-gui</li>
    <li>python312Packages.mim-solvers</li>
    <li>python312Packages.ndcurves</li>
    <li>python312Packages.ndcurves.doc</li>
    <li>python312Packages.neuralfoil</li>
    <li>python312Packages.neuralfoil.dist</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.pinocchio.doc</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite-nlp</li>
    <li>python312Packages.proxsuite-nlp.doc</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.tsid</li>
    <li>python312Packages.tsid.doc</li>
    <li>qgv</li>
    <li>tsid</li>
    <li>tsid.doc</li>
  </ul>
</details>

**Pull request merged**